api/types/device.go

```diff
@@ -233,24 +227,20 @@ func ResourceEnrollStatusToString(enrollStatus devicepb.DeviceEnrollStatus) stri
 		return "enrolled"
 	case devicepb.DeviceEnrollStatus_DEVICE_ENROLL_STATUS_NOT_ENROLLED:
 		return "not_enrolled"
-	case devicepb.DeviceEnrollStatus_DEVICE_ENROLL_STATUS_UNSPECIFIED:
-		return "unspecified"
 	default:
-		return enrollStatus.String()
+		return "unspecified"
 	}
 }
 
 // ResourceEnrollStatusFromString converts a string representation of
 // DeviceEnrollStatus suitable for resource fields to DeviceEnrollStatus.
-func ResourceEnrollStatusFromString(enrollStatus string) (devicepb.DeviceEnrollStatus, error) {
+func ResourceEnrollStatusFromString(enrollStatus string) devicepb.DeviceEnrollStatus {
 	switch enrollStatus {
 	case "enrolled":
-		return devicepb.DeviceEnrollStatus_DEVICE_ENROLL_STATUS_ENROLLED, nil
+		return devicepb.DeviceEnrollStatus_DEVICE_ENROLL_STATUS_ENROLLED
 	case "not_enrolled":
-		return devicepb.DeviceEnrollStatus_DEVICE_ENROLL_STATUS_NOT_ENROLLED, nil
-	case "unspecified":
-		return devicepb.DeviceEnrollStatus_DEVICE_ENROLL_STATUS_UNSPECIFIED, nil
+		return devicepb.DeviceEnrollStatus_DEVICE_ENROLL_STATUS_NOT_ENROLLED
 	default:
```

Comment:
Similarly to above, we are swallowing errors if we see a proto value we don't know about.

Reply:
What would be the consequence of swallowing error here? And otherwise, what purpose would "unspecified" address?

Reply:
Yeah I assumed that. But the purpose of having "unspecified" state is still not clear.

Reply:
Do you mean in general? Calling the zero value enum value "UNSPECIFIED" is a proto best practice: https://protobuf.dev/programming-guides/style/#enums. Ping me on chat if you want further explanation.